src/webhook/validation/oneagent.go

```diff
@@ -90,3 +95,47 @@ func hasConflictingMatchLabels(labelMap, otherLabelMap map[string]string) bool {
 	otherLabelSelectorLabels := labels.Set(otherLabelMap)
 	return labelSelector.Matches(otherLabelSelectorLabels) || otherLabelSelector.Matches(labelSelectorLabels)
 }
+
+func getOneAgentVolumeStorageEnvVar(env []corev1.EnvVar) (bool, bool) {
+	envVar := kubeobjects.FindEnvVar(env, "ONEAGENT_ENABLE_VOLUME_STORAGE")
+	if envVar != nil {
+		return envVar.Value == "true", true
+	}
+	return false, false
+}
+
+func hasOneAgentVolumeStorageEnabled(dynakube *dynatracev1beta1.DynaKube) (bool, bool) {
+	switch {
+	case dynakube.CloudNativeFullstackMode():
+		return getOneAgentVolumeStorageEnvVar(dynakube.Spec.OneAgent.CloudNativeFullStack.Env)
+	case dynakube.ClassicFullStackMode():
+		return getOneAgentVolumeStorageEnvVar(dynakube.Spec.OneAgent.ClassicFullStack.Env)
+	case dynakube.HostMonitoringMode():
+		return getOneAgentVolumeStorageEnvVar(dynakube.Spec.OneAgent.HostMonitoring.Env)
+	}
+	return false, false
+}
+
+func conflictingOneAgentVolumeStorageSettings(dv *dynakubeValidator, dynakube *dynatracev1beta1.DynaKube) string {
+	volumeStorageEnabled, volumeStorageVarSet := hasOneAgentVolumeStorageEnabled(dynakube)
+	if dynakube.NeedsReadOnlyOneAgents() && volumeStorageVarSet && !volumeStorageEnabled {
+		return errorVolumeStorageReadOnlyModeConflict
+	}
+	return ""
+}
+
+func superfluousReadOnlyHostFsFeatureFlag(dv *dynakubeValidator, dynakube *dynatracev1beta1.DynaKube) string {
```

Comment:
```suggestion
func unnecessaryReadOnlyHostFsFeatureFlag(dv *dynakubeValidator, dynakube *dynatracev1beta1.DynaKube) string {
```

Reply:
Agreed, naming is not optimal, but I will even go with "ineffective", that's even more to the point.